components/infobox/wikis/starcraft/infobox_person_player_custom.lua

```diff
@@ -0,0 +1,510 @@
+		placementConditions:add{
+			ConditionNode(ColumnName('placement'), Comparator.eq, item),
+		}
+	end
+
+	local conditions = ConditionTree(BooleanOperator.all):add{
+		playerConditions,
+		ConditionNode(ColumnName('date'), Comparator.neq, '1970-01-01 00:00:00'),
+		ConditionNode(ColumnName('liquipediatiertype'), Comparator.neq, 'Charity'),
+		ConditionTree(BooleanOperator.any):add{
+			ConditionNode(ColumnName('individualprizemoney'), Comparator.gt, '0'),
+			ConditionNode(ColumnName('extradata_award'), Comparator.neq, ''),
+			ConditionTree(BooleanOperator.all):add{
+				ConditionNode(ColumnName('players_type'), Comparator.gt, 'solo'),
+				placementConditions,
+			},
+		},
+	}
+
+	local earnings = {}
+	local medals = {}
+	earnings['total'] = {}
+	medals['total'] = {}
+	local earningsTotal = 0
+
+	local queryParameters = {
+		conditions = conditions:toString(),
+		order = 'weight desc, liquipediatier asc, placement asc',
+	}
+
+	local processPlacement = function(placement)
+		--handle earnings
+		earnings, earningsTotal = CustomPlayer._addPlacementToEarnings(earnings, earningsTotal, placement)
+
+		--handle medals
+		medals = CustomPlayer._addPlacementToMedals(medals, placement)
+	end
+
+	Lpdb.executeMassQuery('placement', queryParameters, processPlacement)
+
+	if #_achievements > 0 then
+		Variables.varDefine('achievements', Json.stringify(_achievements))
+	end
+	if #_awardAchievements > 0 then
+		Variables.varDefine('awardAchievements', Json.stringify(_awardAchievements))
+	end
+	CustomPlayer._setVarsFromTable(earnings)
+	CustomPlayer._setVarsFromTable(medals)
+
+	return earningsTotal, earnings['total']
+end
+
+function CustomPlayer._addPlacementToEarnings(earnings, earningsTotal, data)
+	local mode = _EARNING_MODES[(data.players or {}).type or ''] or 'other'
+	if not earnings[mode] then
+		earnings[mode] = {}
+	end
+	local year = string.sub(data.date, 1, 4)
+	earnings[mode][year] = (earnings[mode][year] or 0) + data.individualprizemoney
+	earnings['total'][year] = (earnings['total'][year] or 0) + data.individualprizemoney
+	earningsTotal = (earningsTotal or 0) + data.individualprizemoney
+
+	return earnings, earningsTotal
+end
+
+function CustomPlayer._addPlacementToMedals(medals, data)
+	if data.liquipediatiertype ~= 'Qualifier' then
+		local place = CustomPlayer._getPlacement(data.placement)
+		CustomPlayer._setAchievements(data, place)
+		if
+			(data.players or {}).type == 'solo'
+			and place and place <= 3
+		then
+			local tier = data.liquipediatier or 'undefined'
+			if not medals[place] then
+				medals[place] = {}
+			end
+			medals[place][tier] = (medals[place][tier] or 0) + 1
+			medals[place]['total'] = (medals[place]['total'] or 0) + 1
+			medals['total'][tier] = (medals['total'][tier] or 0) + 1
+		end
+	end
+
+	return medals
+end
+
+function CustomPlayer._setVarsFromTable(table)
+	for key1, item1 in pairs(table) do
+		for key2, item2 in pairs(item1) do
+			Variables.varDefine(key1 .. '_' .. key2, item2)
+		end
+	end
+end
```

Comment:
Why is it needed? Variable names needs to be improved.

Reply:
basically it sets the wiki vars we get from the medals stuff and from the matchup stuff (once that gets enabled) so that the according statistics display templates on the page can grab it and display it without having to query everything again